src/utils.ts

```diff
@@ -163,5 +163,20 @@ export function mergeOptions(
 }
 
 export function isMoized(fn: Moizeable | Moized | Options): fn is Moized {
-    return typeof fn === 'function' && fn.isMoized;
+    return typeof fn === 'function' && (fn as Moizeable).isMoized;
+}
+
+export function setName(
+    fn: Moized,
+    originalFunctionName: string,
+    profileName: string
+) {
+    const name = originalFunctionName || profileName || 'anonymous';
```

Comment:
I keep debating if `profileName` should be the priority, as it only exists if explicitly provided, and therefore can be considered a "tag" for the function. I won't merge until I get some feedback on this.

Reply:
It seems reasonable to prioritize `profileName` since it has a common purpose of performance profiling. I think it's less important if the original function name can be used by default.